sourmash/commands.py

```diff
@@ -286,33 +457,136 @@ def maybe_add_alignment(alignment, cell_seqs, args, barcodes):
                 notify('calculated {} signatures for {} sequences in {}',
                        len(siglist), n + 1, filename)
             elif args.input_is_10x:
-                import pathos.multiprocessing as mp
-                from .tenx import read_10x_folder
-
-                barcodes, bam_file = read_10x_folder(filename)
-                manager = multiprocessing.Manager()
+                from .tenx import (read_barcodes_file, shard_bam_file,
+                                   bam_to_fasta)
 
-                cell_seqs = manager.dict()
-
-                notify('... reading sequences from {}', filename)
+                # Initializing time
+                startt = time.time()
 
-                pool = mp.Pool(processes=args.processes)
-                pool.map(lambda x: maybe_add_alignment(x, cell_seqs, args, barcodes), bam_file)
-                # for n, alignment in enumerate(bam_file):
-                #     if n % 10000 == 0:
-                #         if n:
-                #             notify('\r...{} {}', filename, n, end='')
-                #     maybe_add_alignment(alignment, cell_seqs)
-
-                cell_signatures = [
-                    build_siglist(seqs, filename=filename, name=barcode)
-                    for barcode, seqs in cell_seqs.items()]
-                sigs = list(itertools.chain(*cell_signatures))
-                if args.output:
-                    siglist += sigs
+                # Setting barcodes file, some 10x files don't have a filtered
+                # barcode file
+                if args.barcodes_file is not None:
+                    barcodes = read_barcodes_file(args.barcodes_file)
                 else:
-                    siglist = sigs
+                    barcodes = None
+
+                # Shard bam file to smaller bam file
+                notify('... reading bam file from {}', filename)
+                n_jobs = args.processes
+                filenames, mmap_file = np_utils.to_memmap(np.array(
+                    shard_bam_file(filename, args.line_count, os.getcwd())))
```

Comment:
are the sharded bam files an output of this command, or should they be temporary (and deleted after the command is done?) if the former, this should be documented somewhere (maybe it is?) if the latter, you should use a context manager here to clean up.

Reply:
they are cleaned up on line 508

Reply:
ahh, ok thx!